### ISSUE TITLE
Not working when connected to multiple I2C devices

### ISSUE THREAD

**Issue by dingdinglhz** [Depau/APDS9930#4]
I have connected the module to an Arduino I2C bus. Although it works if it is the only device in the bus, it doesn't seem to be working when multiple devices (NOT multiple APDS-9900, but multiple I2C devices, one of which is APDS-9900) are connected at the same time. What could be the issue?
It either gives me unexpected values or gives me an reading error.

**Comment by depau** [comment]
@dingdinglhz I don't have time to check this right now. I have a ton of exams to give at the university. However, check:

- that you're using an APDS-9930 and not an APDS-9900 as you mentioned in your post; this library only works with the 9930, but may work with other models with some adaptations (it's actually a fork of the 9960 library)
- that there are no i²c address collisions (in that case you can modify the jumper configuration in the expansion board to select another address; refer to the datasheets)
- that both devices work by themselves (and that the other device works when wired together with another device)
- that the wiring is correct and that any jumper cables or solder are good
- that the Wire (i²c) library you're using is updated: maybe they have a fix for that

On the i²c side, the library uses basic i²c read/write commands. The library is just a wrapper around all the complicated binary commands that you actually need to send to get the sensor up and running.

I'm quite sure it's not related to this library itself but to something else, but I'll leave this issue open as I'm not sure and cannot test at the moment.

If you have the possibility, try the two devices with some other i²c master (e.g a more powerful AVR MCU - Arduino Mega and similar - or a Linux computer: your PC probably does have an i²c port, it's just a bit hidden 😉 http://davideddu.org/blog/posts/graphics-card-i2c-port-howto/ )
I made a Python library for both the APDS 9930 and the i²c LCD backpack for Linux. Check my blog ( https://davideddu.org/blog/posts/apds-9930-python-module/ ) and my GitHub @Depaulicious (python-pywiring + python-liquidcrystal)

Let me know if you find out anything :)

**Comment by dingdinglhz** [comment]
Sorry I forgot to reply, it was quite a while ago actually.
I discovered in the end that it was an address collision. I did not check the address carefully and confused myself between hex and dec representation.
And later I also replaced it with a APDS-9930 (for a different issue), and discovered that these two model have similar interface, but the gives quite different reading values. For my application the 9930 gives more stable and reliable readings.
And thank you very much for the work on this library! ヾ(◍°∇°◍)ﾉﾞ
